diff --git a/node_modules/@nuxtjs/eslint-module/dist/module.js b/node_modules/@nuxtjs/eslint-module/dist/module.js
index 42b583b..0dcfa94 100644
--- a/node_modules/@nuxtjs/eslint-module/dist/module.js
+++ b/node_modules/@nuxtjs/eslint-module/dist/module.js
@@ -22,7 +22,7 @@ function moduleExists(name) {
 const logger = consola__default["default"].withTag("nuxt:eslint");
 const CONFIG_KEY = "eslint";
 const nuxtModule = function(moduleOptions) {
-  if (!this.options.dev) {
+  if (!this || !this.options.dev) {
     return;
   }
   const DEFAULTS = {
