diff --git a/node_modules/@nuxtjs/web-vitals/dist/index.js b/node_modules/@nuxtjs/web-vitals/dist/index.js
index d801823..986779d 100644
--- a/node_modules/@nuxtjs/web-vitals/dist/index.js
+++ b/node_modules/@nuxtjs/web-vitals/dist/index.js
@@ -44,6 +44,7 @@ const PROVIDERS = [
 ];
 
 function webVitalsModule() {
+  if (!this) return;
   const { nuxt } = this;
   const options = defu__default["default"](nuxt.options.webVitals, {
     provider: "",
